.github/workflows/rust.yml

```diff
@@ -36,6 +38,8 @@ jobs:
     - name: Checkout sources
       uses: actions/checkout@v3
 
+    - uses: Swatinem/rust-cache@v2
```

Comment:
additionally: once we set up caching, we need one task per toolchain that runs first and fills the cache. Otherwise, caching is ineffective and produces many GBs of artifacts on GitHub but we only have a 10GB limit. I experienced that in the multiboot2 crate. The pipeline there looks like this:

![Uploading image.png…]()

Reply:
> Otherwise, caching is ineffective and produces many GBs of artifacts on GitHub but we only have a 10GB limit.
simulatnously
My concern here is only that the same cache is uploaded multiple times.. simultaneously by every job as initially, they all do perform the cargo install.